### ISSUE TITLE
For each transaction action, determine which additional data if any is needed to compute net change to value balance

### ISSUE THREAD

**Issue by aubrika** [penumbra-zone/penumbra#1368]
Each transaction action will not necessarily include the data required to compute and display its net effect on the account value balance - survey each action type to determine what additional data is needed as preliminary research for displaying this change.

- [x] `Output`
- [x] `Spend`
- [ ] `Delegate`
- [ ] `Undelegate`
- [ ] `ValidatorDefinition`
- [ ] `IBCAction`
- [x] `Swap`
- [x] `SwapClaim`
- [ ] `ProposalSubmit`
- [ ] `ProposalWithdraw`
- [ ] `DelegatorVote`
- [ ] `ValidatorVote`
- [ ] `PositionOpen`
- [ ] `PositionClose`
- [ ] `PositionWithdraw`
- [ ] `PositionRewardClaim`
- [ ] `ICS20Withdrawal`

**Comment by aubrika** [comment]
During initial design work for #1426, @redshiftzero & I found that currently only Output, Spend, Swap, and SwapClaim Actions will need additional data in the context of a `TransactionView` in order to compute their net effects on account balance. These Action types will require their associated Notes and PayloadKeys to calculate their Amounts, and in the case of Spends, also their Nullifiers. Some of the other Action types above will require additional fields following the implementation of flow encryption, but proximally we should be able to use what we learned to implement the pcli command described in #1364.
